core/src/com/unciv/logic/civilization/ReligionManager.kt

```diff
@@ -84,27 +89,20 @@ class ReligionManager : IsPartOfGameInfoSerialization {
         if (!civInfo.gameInfo.isReligionEnabled()) return false
         if (religionState != ReligionState.None) return false
         if (!civInfo.isMajorCiv()) return false
-        if (civInfo.gameInfo.ruleSet.beliefs.values.none { isPickablePantheonBelief(it) })
-            return false
+        if (numberOfBeliefsAvailable(BeliefType.Pantheon) == 0)
+            return false // no more available pantheons
         if (civInfo.gameInfo.civilizations.any { it.religionManager.religionState == ReligionState.EnhancedReligion })
             return false
         return storedFaith >= faithForPantheon()
+                || (freeBeliefs[BeliefType.Pantheon] != null && freeBeliefs[BeliefType.Pantheon]!! > 0)
     }
 
-    fun isPickablePantheonBelief(belief: Belief): Boolean {
-        if (belief.type != BeliefType.Pantheon) return false
-        return (civInfo.gameInfo.civilizations.none { it.religionManager.religion?.followerBeliefs?.contains(belief.name) == true })
-    }
-
-    fun choosePantheonBelief(belief: Belief) {
-        storedFaith -= faithForPantheon()
+    private fun choosePantheonBelief(belief: Belief) {
```

Comment:
I'm not sure this function name still reflects its use, maybe change it to `foundPantheon`. In that case, you might as well change the `belief: Belief` parameter to `religionName: string`, as that is the only thing the `belief` actually used for in this function